fastestimator/trace/metric/dice.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2019 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from typing import Optional
+
+import numpy as np
+
+from fastestimator.backend.to_number import to_number
+from fastestimator.trace.trace import Trace
+from fastestimator.util import Data
+
+
+class Dice(Trace):
+    """ Dice score for binary classification between y_true and y_predict.
+
+    Args:
+        true_key: The key of the ground truth mask
+        pred_key: The key of the prediction values
+        mode: What mode to execute in (None to execute in all modes)
+        output_name: What to call the output from this trace (for example in the logger output)
+    """
+    def __init__(self,
+                 true_key: str,
+                 pred_key: str,
+                 threshold: float = 0.5,
+                 mode: Optional[str] = "eval",
```

Comment:
could be list of strings also

Reply:
but also, default should be ("eval", "test") I think

Reply:
fixed